test_app/authentication/service_token_auth.py

```diff
@@ -0,0 +1,39 @@
+import jwt
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from rest_framework.authentication import BaseAuthentication, get_authorization_header
+
+from ansible_base.resource_registry.resource_server import get_resource_server_config
+
+User = get_user_model()
+
+
+class ServiceTokenAuthentication(BaseAuthentication):
+    keyword = "Token"
+
+    def authenticate(self, request):
+        auth = get_authorization_header(request).split()
+
+        if not auth or auth[0].lower() != self.keyword.lower().encode():
+            return None
+
+        token = auth[1]
```

Comment:
Do we need to worry about a malformed authoriation header? i.e. could this only every be length of 2 or is length 1 possible?

Reply:
Honestly, I just copied this from the DRF token auth class. I can double check that my implementation is correct.